pkg/executor/util.go

```diff
@@ -24,6 +24,8 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+const nameLengthLimit = 52
```

Comment:
We probably need comments that have a detailed breakdown of how this limit is reached. I did some noodling myself to see if I could figure it out:

Max statefulset/pod/pvc name size is 253. Max service name size is 63, so that will be our bottleneck.

We use "crname-roleid" as the base prefix for creating the statefulset name, which then leads to the other names.

Let's say we're using GenerateName for the statefulset name (as per the other comments). GenerateName suffix adds 5 (usually/atleast) ... will eat into end of name if necessary, but let's try to avoid that. We'll probably want to add a dash at the end of our prefix in order to nicely separate it from the generated UID, so the whole cost of GenerateName would be 6 characters.

The service name will be statefulset-name plus dash plus index number. The index number could be up to 4 digits currently (cf. maxKDMembers). So the cost of those suffixes on the service name is 5 characters.

Therefore the length of the "crname-roleid" prefix + 5 + 6 must be <= 63. I.e. the length of the prefix must be <= 52.

Since this does depend on the number of digits in maxKDMembers, we probably want to either assert at runtime that maxKDMembers has no more than 4 digits, or else calculate the number of digits on the fly when creating this limit. If/when maxKDMembers becomes configurable that will be even more important to get right.

Reply:
Is pod name, statefulset max, pvc name max sizes documented somewhere? I just considered 63 as max for everything and came about with the number 52 with some experimentation. I will document this.

Reply:
This is where I was trying to get a conversation started because the setting the value to 52 with the assumption that pod name would contain cr name + role name + UID + index, I could never see it in the kubectl get pods output. Bizarre! I know. I tried experimenting a lot and landed up at number 46. I don't have a logical explanation as to why that is the case.

Reply:
You couldn't actually see any pods, or you saw pods that had a different name?

I can try building/deploying this branch myself to see what is going on.

Reply:
I couldn't actually see pods.

Reply:
So if a statefulset doesn't generate pods, there's something amiss in its pod template spec. For example https://github.com/bluek8s/kubedirector/issues/387 would manifest in this way. IIRC you can see what's going on by doing a kubectl describe on the statefulset (to see its events).